### ISSUE TITLE
Template not working with latest React Native CLI

### ISSUE THREAD

**Issue by idkjs** [reasonml-community/reason-react-native#570]
Got error when running `npx react-native init ReasonmlDemo --template reason`

```reason
DEMOS npx react-native init ReasonmlDemo --template reason

               ######                ######
             ###     ####        ####     ###
            ##          ###    ###          ##
            ##             ####             ##
            ##             ####             ##
            ##           ##    ##           ##
            ##         ###      ###         ##
             ##  ########################  ##
          ######    ###            ###    ######
      ###     ##    ##              ##    ##     ###
   ###         ## ###      ####      ### ##         ###
  ##           ####      ########      ####           ##
 ##             ###     ##########     ###             ##
  ##           ####      ########      ####           ##
   ###         ## ###      ####      ### ##         ###
      ###     ##    ##              ##    ##     ###
          ######    ###            ###    ######
             ##  ########################  ##
            ##         ###      ###         ##
            ##           ##    ##           ##
            ##             ####             ##
            ##             ####             ##
            ##          ###    ###          ##
             ###     ####        ####     ###
               ######                ######


                  Welcome to React Native!
                 Learn Once Write Anywhere

✔ Downloading template
✖ Copying template
error Error: Cannot find module '/var/folders/dz/xzttny_14gs2lgxcvl5lm5680000gn/T/rncli-init-template-fWzoC3/node_modules/reason/template.config'
Require stack:
- /Users/prisc_000/.npm/_npx/47740/lib/node_modules/react-native/node_modules/@react-native-community/cli/build/commands/init/template.js
- /Users/prisc_000/.npm/_npx/47740/lib/node_modules/react-native/node_modules/@react-native-community/cli/build/commands/init/init.js
- /Users/prisc_000/.npm/_npx/47740/lib/node_modules/react-native/node_modules/@react-native-community/cli/build/commands/init/index.js
- /Users/prisc_000/.npm/_npx/47740/lib/node_modules/react-native/node_modules/@react-native-community/cli/build/commands/index.js
- /Users/prisc_000/.npm/_npx/47740/lib/node_modules/react-native/node_modules/@react-native-community/cli/build/cliEntry.js
- /Users/prisc_000/.npm/_npx/47740/lib/node_modules/react-native/node_modules/@react-native-community/cli/build/index.js
- /Users/prisc_000/.npm/_npx/47740/lib/node_modules/react-native/cli.js
```

**Comment by MoOx** [comment]
It seems we didn't prepare the template for the new react native CLI.

**Comment by wokalski** [comment]
@MoOx the typescript template isn't ready either, no worries :)

**Comment by idkjs** [comment]
https://github.com/reasonml-community/reason-react-native/pull/572

**Comment by MoOx** [comment]
Keeping open until PR is merged

(Reminder: the PR contains `Closes #570` so as soon as the commit is merged into master, this issue will be automatically closed)

**Comment by idkjs** [comment]
Nice. Did not know that. Thank you.
